src/main/java/javaracingcar/domain/Car.java

```diff
@@ -0,0 +1,51 @@
+package javaracingcar.domain;
+
+public class Car {
+    private String name;
+    private int position;
+
+    private Car(String name) {
+
+        this.name = name;
+        this.position = 0;
+    }
+
+    public String getName() {
+        return name;
+    }
+
+    public int getPosition() {
+        return position;
+    }
+
+    private static final int CAR_NAME_LENGTH_MAX_SIZE = 4;
+
+    public static Car generateCar(String name) {
+        validateNull(name);
+        validateEmptyString(name);
+        validateNameLength(name);
+        return new Car(name);
+    }
+
+    private static void validateNull(String name) {
+        if (name == null) {
+            throw new IllegalArgumentException("null이 입력되었습니다."); //TODO enum으로 변경
+        }
+    }
+
+    private static void validateEmptyString(String name) {
+        if (name.equals("")) {
```

Comment:
name에 null이 들어올 경우 NPE가 터지는데요 (물론 null 체크를 먼저 하고 있지만 메서드 순서가 바뀌거나 하면 문제가 생기겠네요)
두 대상의 동등성을 비교할 때에는 더 확실한 쪽을 기준으로 equals를 호출하는 것이 좋습니다 :)
```suggestion
        if ("".equals(name)) {
```